integrations/azure_ai_search/src/haystack_integrations/document_stores/azure_ai_search/document_store.py

```diff
@@ -0,0 +1,450 @@
+            }
+
+            # Create the document with meta only if it's non-empty
+            doc = Document(
+                id=azure_doc["id"], content=azure_doc["content"], embedding=embedding, meta=meta if meta else {}
+            )
+
+            documents.append(doc)
+        return documents
+
+    def index_exists(self, index_name: Optional[str]) -> bool:
+        """
+        Check if the index exists in the Azure AI Search service.
+
+        :param index_name: The name of the index to check.
+        :returns bool: whether the index exists.
+        """
+
+        if self._index_client and index_name:
+            return index_name in self._index_client.list_index_names()
+        else:
+            msg = "Index name is required to check if the index exists."
+            raise ValueError(msg)
+
+    def _get_raw_documents_by_id(self, document_ids: List[str]):
+        """
+        Retrieves all Azure documents with a matching document_ids from the document store.
+
+        :param document_ids: ids of the documents to be retrieved.
+        :returns: list of retrieved Azure documents.
+        """
+        azure_documents = []
+        for doc_id in document_ids:
+            try:
+                document = self.client.get_document(doc_id)
+                azure_documents.append(document)
+            except ResourceNotFoundError:
+                logger.warning(f"Document with ID {doc_id} not found.")
+        return azure_documents
+
+    def _convert_haystack_documents_to_azure(self, document: Dict[str, Any]) -> Dict[str, Any]:
+        """Map the document keys to fields of search index"""
+
+        # Because Azure Search does not allow dynamic fields, we only include fields that are part of the schema
+        index_document = {k: v for k, v in {**document, **document.get("meta", {})}.items() if k in self._index_fields}
+        if index_document["embedding"] is None:
+            index_document["embedding"] = self._dummy_vector
+
+        return index_document
+
+    def _create_metadata_index_fields(self, metadata: Dict[str, Any]) -> List[SimpleField]:
+        """Create a list of index fields for storing metadata values."""
+
+        index_fields = []
+        metadata_field_mapping = self._map_metadata_field_types(metadata)
+
+        for key, field_type in metadata_field_mapping.items():
+            index_fields.append(SimpleField(name=key, type=field_type, filterable=True))
+
+        return index_fields
+
+    def _map_metadata_field_types(self, metadata: Dict[str, type]) -> Dict[str, str]:
+        """Map metadata field types to Azure Search field types."""
+
+        metadata_field_mapping = {}
+
+        for key, value_type in metadata.items():
+
+            # Azure Search index only allows field names starting with letters
+            field_name = next((key[i:] for i, char in enumerate(key) if char.isalpha()), key)
```

Comment:
I would probably raise an error...

Reply:
Related issue discussed [here](https://deepset-ai.slack.com/archives/C06FBMD8LMV/p1730376446186879). For now will just an error.

Reply:
In general, I would not rename fields.
In case some names are not supported, let's emit a clear warning and drop them.